lib/ReactViews/Preview/DataPreviewMap.tsx

```diff
@@ -0,0 +1,334 @@
+
+  const mapContainerRef = useRef<HTMLDivElement>(null);
+
+  const isZoomedToExtentRef = useRef(false);
+
+  const homeCamera = useMemo(() => {
+    return terria.mainViewer.homeCamera;
+  }, [terria]);
+
+  const boundingRectangleCatalogItem = useMemo(() => {
+    const rectangle = previewed?.rectangle;
+    if (!rectangle) {
+      return undefined;
+    }
+    const { west, south, east, north } = rectangle;
+
+    if (
+      west === undefined ||
+      south === undefined ||
+      east === undefined ||
+      north === undefined
+    ) {
+      return undefined;
+    }
+
+    return createBoundingRectangleCatalogItem(
+      terria,
+      {
+        west,
+        south,
+        east,
+        north
+      },
+      "__preview-data-extent"
+    );
+  }, [previewed?.rectangle, terria]);
+
+  const zoomedOutBoundingRectangleCatalogItem = useMemo(() => {
+    const rectangle = previewed?.rectangle;
+    if (!rectangle) {
+      return undefined;
+    }
+    let { west, south, east, north } = rectangle;
+
+    if (
+      west === undefined ||
+      south === undefined ||
+      east === undefined ||
+      north === undefined
+    ) {
+      return undefined;
+    }
+
+    // When zoomed out, make sure the dataset rectangle is at least 5% of the width and height
+    // the home view, so that it is actually visible.
+    const minimumFraction = 0.05;
+    const homeView = homeCamera;
+    const minimumWidth =
+      CesiumMath.toDegrees(homeView.rectangle.width) * minimumFraction;
+    if (east - west < minimumWidth) {
+      const center = (east + west) * 0.5;
+      west = center - minimumWidth * 0.5;
+      east = center + minimumWidth * 0.5;
+    }
+
+    const minimumHeight =
+      CesiumMath.toDegrees(homeView.rectangle.height) * minimumFraction;
+    if (north - south < minimumHeight) {
+      const center = (north + south) * 0.5;
+      south = center - minimumHeight * 0.5;
+      north = center + minimumHeight * 0.5;
+    }
+
+    return createBoundingRectangleCatalogItem(
+      terria,
+      {
+        west,
+        south,
+        east,
+        north
+      },
+      "__preview-data-extent-zoomed-out"
+    );
+  }, [homeCamera, previewed?.rectangle, terria]);
+
+  const previewViewer = useMemo(() => {
+    const viewer = new TerriaViewer(
+      terria,
+      mobxComputed(() => {
+        const previewItem = new AdaptForPreviewMap(undefined, terria);
+        previewItem.previewed = previewed;
+        return filterOutUndefined([
+          previewItem,
+          boundingRectangleCatalogItem,
+          zoomedOutBoundingRectangleCatalogItem
+        ]);
+      })
+    );
+
+    viewer.viewerMode = ViewerMode.Leaflet;
+    viewer.disableInteraction = true;
+    viewer.homeCamera = homeCamera;
+
+    return viewer;
+  }, [
+    boundingRectangleCatalogItem,
+    homeCamera,
+    previewed,
+    terria,
+    zoomedOutBoundingRectangleCatalogItem
+  ]);
+
+  useEffect(() => {
+    console.log(
```

Comment:
Is the `console.log` leftover debug code?

Reply:
This is from the existing codebase, probably was added to avoid future regressions. Will see if it can be ported to tests